developers/blobstream-rollups.md

```diff
@@ -0,0 +1,123 @@
+
+### Optimistic rollups that uses a sequence of spans
+
+One way to construct an optimistic rollup that uses Celestia as a DA layer is to post the rollup data in Celestia.
+Then in the rollup header, use the following information to reference a sequence of spans, aka a data pointer:
+
+- `Height`: the height of the Celestia block containing the rollup data
+- `Start share`: the index of the first share containing the rollup data
+- `Data length`: the number of shares that the rollup data occupies
+
+Note:
+If the rollup data is submitted in multiple blocks,
+the above sequence of spans can be generalized to include multiple blocks.
+For simplicity, we will stick with the data only submitted to a single Celestia block.
+
+Now that this information is saved in the rollup header/settlement contract,
+and users/rollup full nodes will be able to query for the rollup data from Celestia and verify that it's valid.
+And if it's not, fraud proofs can be generated.
+
+The proofs in this setup are discussed in the [inclusion proofs](https://github.com/celestiaorg/blobstream-contracts/blob/master/docs/inclusion-proofs.md) documentation.
+
+### Optimistic rollups that use share commitments
+
+Another way to build a rollup is to replace the sequence of spans with a `height` and a `share commitment`.
+Then, users/rollup full nodes will be able to query that data, and in case of a dispute, they can create a fraud proof.
+
+The difference between the above construction and this one is that the proofs used will be different:
+when using a sequence of spans,
+the inclusion proofs will be straight from the rollup data up to the [data root tuple root](https://github.com/celestiaorg/blobstream-contracts/blob/master/README.md#how-it-works).
+However, in the case of using the `height` and the `share commitment`,
+an extra step would need to be done when posting the header to the settlement contract.
+This step is proving that the provided commitment is part of the Celestia block (referenced by its height).
+Then, fraud proofs will need to prove the following:
+
+- share inclusion to the share commitment: meaning creating two merkle proofs:
+  - share merkle proof up to the [subtree root](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html#blob-share-commitment-rules)
+    corresponding to that share
+  - subtree root merkle proof to the [share commitment](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html#blob-share-commitment-rules)
+- share commitment inclusion to the data root tuple root: meaning four merkle proofs:
+  - subtree roots merkle proofs to the share commitment: to make sure the subtree roots are valid
+  - subtree roots merkle proofs up to the row roots: to prove that the subtree roots belong to a set of rows
+  - row roots proofs to the data root
+  - data root tuple proof to the data root tuple
+
+Once these are valid, the rollup contract can proceed to parse the share and verify the contested claim.
```

Comment:
"contested claim" is an undefined phrase, I'm not sure what that means. Is that a fraud proof?